crates/ruff_linter/src/rules/refurb/rules/isinstance_type_none.rs

```diff
@@ -88,29 +85,63 @@ fn is_none(expr: &Expr) -> bool {
             // Ex) `type(None)`
             Expr::Call(ast::ExprCall {
                 func, arguments, ..
-            }) if arguments.len() == 1 => {
-                if let Expr::Name(ast::ExprName { id, .. }) = func.as_ref() {
-                    if id.as_str() == "type" {
-                        return matches!(arguments.args.first(), Some(Expr::NoneLiteral(_)));
-                    }
+            }) => {
+                if !semantic.match_builtin_expr(func, "type") {
+                    return false;
                 }
-                false
+
+                matches!(arguments.args.as_ref(), [Expr::NoneLiteral(_)])
             }
 
             // Ex) `(type(None),)`
-            Expr::Tuple(tuple) => tuple.iter().all(|element| inner(element, false)),
+            Expr::Tuple(tuple) => tuple.iter().all(|element| inner(element, false, semantic)),
 
             // Ex) `type(None) | type(None)`
             Expr::BinOp(ast::ExprBinOp {
                 left,
                 op: Operator::BitOr,
                 right,
                 ..
-            }) => inner(left, true) && inner(right, true),
+            }) => {
+                // `None | None` is a `TypeError` at runtime
+                if left.is_none_literal_expr() && right.is_none_literal_expr() {
+                    return false;
+                }
+
+                inner(left, true, semantic) && inner(right, true, semantic)
+            }
 
             // Otherwise, return false.
             _ => false,
         }
     }
-    inner(expr, false)
+    inner(expr, false, semantic)
+}
+
+fn replace_with_identity_check(expr: &Expr, range: TextRange, checker: &mut Checker) -> Fix {
```

Comment:
This doesn't need to be `&mut`, since you don't use any methods on `Checker` in this function that require a mutable reference

```suggestion
fn replace_with_identity_check(expr: &Expr, range: TextRange, checker: &Checker) -> Fix {
```